forensics/Copy-VmDigitalEvidence.ps1

```diff
@@ -42,152 +42,190 @@ $ErrorActionPreference = 'Stop'
+    Start-AzStorageFileCopy -AbsoluteUri $snapSasUrl.AccessSAS -DestShareName $destTempShare -DestContext $targetStorageContextFile -DestFilePath $SnapshotName -Force
 
-    $ddiskpath = "$targetLinuxDir/$dsnapshotName"
-    
-    $dhashfull = Invoke-Expression -Command "sha256sum $ddiskpath"
-    $dhash = $dhashfull.split(" ")[0]
+    Write-Output "Waiting Fileshare Copy End"
+    Get-AzStorageFileCopyState -Context $targetStorageContextFile -ShareName $destTempShare -FilePath $SnapshotName -WaitForComplete
 
-    Write-Output "Computed SHA-256: $dhash"
+    # Linux hash version if you use a Linux Hybrid Runbook Worker
+    $diskpath = "$targetLinuxDir/$snapshotName"    
+    Write-Output "Start Calculating HASH for $diskpath"
+    $hashfull = Invoke-Expression -Command "sha256sum $diskpath" 
+    $hash = $hashfull.split(" ")[0]
+    Write-Output "Computed SHA-256: $hash"
 
-    Start-AzStorageBlobCopy -AbsoluteUri $dsnapSasUrl.AccessSAS -DestContainer $destSAContainer -DestContext $targetStorageContext -DestBlob $dsnapshotName -Force
-    
-    $BEKurl = $ddisk.EncryptionSettingsCollection.EncryptionSettings.DiskEncryptionKey.SecretUrl
+    #################### Copy the OS BEK to the SOC Key Vault  ###################################
+    $BEKurl = $disk.EncryptionSettingsCollection.EncryptionSettings.DiskEncryptionKey.SecretUrl
     Write-Output "#################################"
-    Write-Output "Disk Encryption Secret URL: $BEKurl"
+    Write-Output "OS Disk Encryption Secret URL: $BEKurl"
     Write-Output "#################################"
     if ($BEKurl) {
         $sourcekv = $BEKurl.Split("/")
         $BEK = Get-AzKeyVaultSecret -VaultName  $sourcekv[2].split(".")[0] -Name $sourcekv[4] -Version $sourcekv[5]
         Write-Output "Key value: $BEK"
-        Write-Output "Secret name: $dsnapshotName"
-        Set-AzKeyVaultSecret -VaultName $destKV -Name $dsnapshotName -SecretValue $BEK.SecretValue -ContentType "BEK" -Tag $BEK.Tags
+        $BEK.Tags.Hash = $hash
+        Get-AzSubscription -SubscriptionId $destSubId | Set-AzContext
+        $secretName = $snapshotName.Replace("_","-")
+        Set-AzKeyVaultSecret -VaultName $destKV -Name $secretName -SecretValue $BEK.SecretValue -ContentType "BEK" -Tag $BEK.Tags
     }
-    
+
+
+    ######## Copy the OS disk hash value in key vault and delete disk in file share ##################
     Write-Output "#################################"
-    Write-Output "Data disk - Put hash value in Key Vault"
+    Write-Output "OS disk - Put hash value in Key Vault"
     Write-Output "#################################"
-    $Secret = ConvertTo-SecureString -String $dhash -AsPlainText -Force
-    Set-AzKeyVaultSecret -VaultName $destKV -Name "$dsnapshotName-sha256" -SecretValue $Secret
+    $secret = ConvertTo-SecureString -String $hash -AsPlainText -Force
+    Set-AzKeyVaultSecret -VaultName $destKV -Name "$SnapshotName-sha256" -SecretValue $secret -ContentType "HASH"

```

Comment:
Idea: If we're going to set `ContentType`, I'd propose setting it to a mime type.  The fact that it's a hash is pretty clear by the secret name of `<snapshot>-sha256`

```suggestion
    Set-AzKeyVaultSecret -VaultName $destKV -Name "$SnapshotName-sha256" -SecretValue $secret -ContentType "text/plain"
```

Reply:
done

Reply:
👍